prov/efa/src/rdm/efa_rdm_pke.c

```diff
@@ -392,70 +386,65 @@ ssize_t efa_rdm_pke_sendv(struct efa_rdm_ep *ep,
 	conn = efa_av_addr_to_conn(ep->base_ep.av, pkt_entry_vec[0]->addr);
 	assert(conn && conn->ep_addr);
 
-	prev_send_wr = NULL;
-	for (i = 0; i < pkt_entry_cnt; ++i) {
-		pkt_entry = pkt_entry_vec[i];
+	qp = ep->base_ep.qp;
+	ibv_wr_start(qp->ibv_qp_ex);
+	for (pkt_idx = 0; pkt_idx < pkt_entry_cnt; ++pkt_idx) {
+		pkt_entry = pkt_entry_vec[pkt_idx];
 		assert(pkt_entry->pkt_size);
 		assert(efa_rdm_ep_get_peer(ep, pkt_entry->addr) == peer);
 
 		send = pkt_entry->send;
 		/* EFA device supports a maximum of 2 iov/SGE */
 		assert(send && send->iov_count <= 2);
-		if (send->iov_count == 0) {
+		if (send->iov_count < 2) {
```

Comment:
This seems like a weird branch.  I think it must be because of how pkt_entry was set up, but I'm surprised that iov_count==1 case doesn't do the loop below.  I see that previously this branch was only taken when iov_count==0, why did that change to ==0 or ==1?

Reply:
Yes. A follow up commit will clean up the logic and stop using the `rxr_pkt_sendv` structure.

Reply:
There is an upcoming commit that cleans this logic up.